src/main/kotlin/com/example/passmanager/repositories/PassRepository.kt

```diff
@@ -1,12 +1,19 @@
 package com.example.passmanager.repositories
 
 import com.example.passmanager.domain.MongoPass
-import com.example.passmanager.domain.MongoPassOwner
-import org.bson.types.ObjectId
-import org.springframework.data.mongodb.repository.MongoRepository
-import java.time.Instant
+import com.example.passmanager.web.dto.PriceDistribution
+import java.math.BigDecimal
+import java.time.LocalDate
 
-interface PassRepository : MongoRepository<MongoPass, ObjectId> {
-    fun findAllByPassOwnerAndPurchasedAtGreaterThan(mongoPassOwner: MongoPassOwner, afterDate: Instant): List<MongoPass>
-    fun findAllByPassOwnerId(passOwnerId: ObjectId): List<MongoPass>
+interface PassRepository {
+    fun findByOwnerAndPurchasedAfter(passOwnerId: String, afterDate: LocalDate): List<MongoPass>
+    fun findAllByPassOwnerId(passOwnerId: String): List<MongoPass>
+    fun findById(passId: String): MongoPass?
+    fun insert(newPass: MongoPass): MongoPass
+    fun save(pass: MongoPass): MongoPass
+    fun deleteById(passId: String)
+    fun deleteAllByOwnerId(passOwnerId: String)
+    fun deleteByIdAndOwnerId(passId: String, passOwnerId: String): Boolean
+    fun sumPurchasedAtAfterDate(passOwnerId: String, afterDate: LocalDate): BigDecimal
+    fun getPassesPriceDistribution(passOwnerId: String): List<PriceDistribution>
```

Comment:
ProceDistribution is a Mongo object? If so please name it with Mongo- prefix

Reply:
No, it is not Mongo object, it is simple data class to store aggregation results

`
data class PriceDistribution(
    val typeName: String?,
    val spentForPassType: BigDecimal)`